src/cursor/cur_backup.c

```diff
@@ -333,6 +333,12 @@ __backup_add_id(WT_SESSION_IMPL *session, WT_CONFIG_ITEM *cval)
 
     conn = S2C(session);
     blk = NULL;
+
+    /* Return error if block-based incremental backup is performed with open LSM trees.  */
+    if (!TAILQ_EMPTY(&conn->lsmqh)) {
+        WT_RET_MSG(session, ENOTSUP, "LSM does not work with block-based incremental backup");
+    }
```

Comment:
I think this should move (and use the `incremental_config` setting) to the compatibility testing chunk of code in `__backup_config` in this file.

Reply:
Moved to check the condition to  `__backup_config ` function.